apps/emqx_bridge_kafka/src/emqx_bridge_kafka.erl

```diff
@@ -641,17 +627,24 @@ kafka_producer_converter(undefined, _HoconOpts) ->
 kafka_producer_converter(
     #{<<"producer">> := OldOpts0, <<"bootstrap_hosts">> := _} = Config0, _HoconOpts
 ) ->
-    %% old schema
+    %% older schema (before 5.1.0)
     MQTTOpts = maps:get(<<"mqtt">>, OldOpts0, #{}),
     LocalTopic = maps:get(<<"topic">>, MQTTOpts, undefined),
-    KafkaOpts = maps:get(<<"kafka">>, OldOpts0),
+    KafkaOpts = maps:get(<<"producer">>, OldOpts0),
```

Comment:
Q: this shouldn't change, right?  Since the older schema apparently had a `kafka` key inside `producer` since long ago?